docs/sources/flow/reference/components/otelcol.processor.attributes.md

```diff
@@ -0,0 +1,582 @@
+    }
+    action {
+        key = "credit_card"
+        action = "delete"
+    }
+    action {
+        key = "duplicate_key"
+        action = "delete"
+    }
+    output {
+        metrics = [otelcol.exporter.otlp.default.input]
+        logs    = [otelcol.exporter.otlp.default.input]
+        traces  = [otelcol.exporter.otlp.default.input]
+    }
+}
+```
+
+### Excluding spans based on resources
+
+A "strict" `match_type` means that we must match the `library` key/value pairs strictly.
+
+```river
+otelcol.processor.attributes "default" {
+    exclude {
+        match_type = "strict"
+        library {
+            name = "mongo-java-driver"
+            version = "3.8.0"
+        }
+    }
+    action {
+        key = "credit_card"
+        action = "delete"
+    }
+    action {
+        key = "duplicate_key"
+        action = "delete"
+    }
+    output {
+        metrics = [otelcol.exporter.otlp.default.input]
+        logs    = [otelcol.exporter.otlp.default.input]
+        traces  = [otelcol.exporter.otlp.default.input]
+    }
+}
+```
+
+### Including and excluding spans based on regex and services
+
+This processor will remove the "token" attribute and will obfuscate the "password" attribute 
+in spans where the service name matches `"auth.*"` and where the span name does not match `"login.*"`.
+
+```river
+otelcol.processor.attributes "default" {
+    // Specifies the span properties that must exist for the processor to be applied.
+    include {
+        // "match_type" defines that "services" is an array of regexp-es.
+        match_type = "regexp"
+        // The span service name must match "auth.*" pattern.
+        services = ["auth.*"]
+    }
+
+    exclude {
+        // "match_type" defines that "span_names" is an array of regexp-es.
+        match_type = "regexp"
+        // The span name must not match "login.*" pattern.
+        span_names = ["login.*"]
+    }
+
+    action {
+        key = "password"
+        action = "update"
+        value = "obfuscated"
+    }
+
+    action {
+        key = "token"
+        action = "delete"
+    }
+
+    output {
+        metrics = [otelcol.exporter.otlp.default.input]
+        logs    = [otelcol.exporter.otlp.default.input]
+        traces  = [otelcol.exporter.otlp.default.input]
+    }
+}
+```
+
+### Including spans based on regex and attributes
+
+The following demonstrates how to process spans that have an attribute that matches a regexp patterns.
+This processor will obfuscate the "db.statement" attribute in spans where the "db.statement" attribute
+matches a regex pattern.
+
+```river
+otelcol.processor.attributes "default" {
+    include {
+        // "match_type" of "regexp" defines that the "value" attributes 
+        // in the "attribute" blocks are regexp-es.
+        match_type = "regexp"
+
+        // This attribute ('db.statement') must exist in the span and match 
+        // the regex ('SELECT \* FROM USERS.*') for a match.
+        attribute {
+            key = "env"
```

Comment:
Shouldn't key = "env" be key = "db.statement"? This was taken from the [examples](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.63.0/processor/attributesprocessor/testdata/config.yaml#L303) in the otel repo.